src/parser.cpp

```diff
@@ -520,6 +520,15 @@ namespace ada::parser {
     // Let pointer be a pointer for input.
     std::string_view::iterator pointer = pointer_start;
 
+    // TODO: Search from right to left, and find the last occurrence.
+    std::string_view::iterator fragment_pointer_start = std::find_if(pointer_start, pointer_end, [](char c) { return c == '#'; });
+
+    if (fragment_pointer_start != pointer_end) {
+      url.fragment = unicode::percent_encode(std::string(fragment_pointer_start + 1, pointer_end),
```

Comment:
Here, until you need to make a copy to percent_encode, I would just pass a string_view... so std::string_view(fragment_pointer_start, pointer_end-fragment_pointer_start-1) or something like it. The compiler may elide the copy, but maybe not. If it can't elide the copy, then it may even end up doing heap allocation!!!

Reply:
This somehow made the code 500ns slower on my local machine:

```
BasicBench_AdaURL            9627 ns         9627 ns        73035 time/byte=11.9145ns
```